pkgs/development/compilers/terra/default.nix

```diff
@@ -27,11 +35,16 @@ in stdenv.mkDerivation rec {
   '';
 
   installPhase = ''
-    mkdir -p $out
+    mkdir -p $out $bin $dev
     cp -r "release/"* $out
-  '';
+    mv $out/lib $dev
```

Comment:
`dev` typically contains only headers and pkgconfig files, i.e., stuff you only need when building stuff. If you want a separate output for shared objects, use `lib` for that. I don't think it makes sense to have both `lib` and `bin` outputs. I'd keep the binary in `out` and have separate `lib` and `dev`, to allow installing them without bringing in the interpreter binary.